### PR TITLE
Align inAppWallet secret sharing implementation with web and previous RN SDK

### PR DESCRIPTION
**Pull request by joaquim-verges**
## PR-Codex overview
The focus of this PR is to align the inAppWallet secret sharing implementation with web and previous RN SDK.

### Detailed summary
- Updated secret sharing implementation in `sss.ts`
- Modified encryption logic in `encryption.ts`
- Refactored key splitting in `creation.ts`
- Improved private key handling in `retrieval.ts`

> The following files were skipped due to too many changes: `packages/thirdweb/src/wallets/in-app/native/helpers/wallet/sss.ts`, `pnpm-lock.yaml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

---

## PR-Codex overview
This PR aligns the in-app wallet secret sharing implementation with web and previous React Native SDK. 

### Detailed summary
- Added `sss.ts` file for secret sharing in the wallet
- Updated error messages in various files
- Improved private key handling in wallet creation
- Enhanced random generation function in `sss.ts`

> The following files were skipped due to too many changes: `packages/thirdweb/src/wallets/in-app/native/helpers/wallet/sss.ts`, `pnpm-lock.yaml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

---

## PR-Codex overview
This PR aligns the in-app wallet secret sharing implementation across platforms. 

### Detailed summary
- Added `sss.ts` in `wallets/in-app/native/helpers/wallet`
- Updated error messages in various files
- Refactored private key handling in `creation.ts` and `retrieval.ts`
- Modified `sss.ts` for secret sharing functionality

> The following files were skipped due to too many changes: `packages/thirdweb/src/wallets/in-app/native/helpers/wallet/sss.ts`, `pnpm-lock.yaml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by joaquim-verges** [comment]
* **#3271** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3271?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @joaquim-verges and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by joaquim-verges — COMMENTED** [review]
/release-pr

File: .changeset/rude-hounds-type.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@thirdweb-dev/react-native-adapter": minor
+"thirdweb": minor
```
[the remark this answers is not in the document]
Comment (joaquim-verges) [reply .changeset/rude-hounds-type.md:3]: it actually breaks previous accounts created with the previous secrets library, so thught minor was appropriate

[the remark this answers is not in the document]
Comment (jnsdls) [reply .changeset/rude-hounds-type.md:3]: ok

**Review by jnsdls — APPROVED** [review]

**Comment by joaquim-verges** [comment]
### Merge activity

* **Jun 11, 7:43 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/thirdweb-dev/js/3271) couldn't add this PR to the GitHub merge queue because it had conflicts with the trunk branch.

**Pull request merged**